### PR TITLE
Feature/s3 c 3118 flatten objmd retention

### PR DESCRIPTION
**Pull request by dora-korpar**
Refactors APIs for flattening of ObjectMD retention object

### CODE REVIEW

File: lib/api/objectPutRetention.js [hunk 1]
```diff
@@ -79,14 +79,14 @@ function objectPutRetention(authInfo, request, log, callback) {
                 (err, retentionInfo) => next(err, bucket, retentionInfo, objectMD));
         },
         (bucket, retentionInfo, objectMD, next) => {
-            // eslint-disable-next-line no-param-reassign
-            objectMD.retentionInfo = retentionInfo;
+            /* eslint-disable no-param-reassign */
```
Comment (jonathan-gramain) [inline lib/api/objectPutRetention.js:82]: To be clean, you should re-enable this warning down in the code as soon as you don't need it for your specific purpose. (whether the linter warning is useful or not is another story :wink:)

**Review by jonathan-gramain — APPROVED** [review]

**Comment by dora-korpar** [comment]
@bert-e reset

**Review by ilkescality — APPROVED** [review]

File: lib/services.js [hunk 2]
```diff
@@ -390,8 +391,9 @@ const services = {
         if (params.tagging) {
             multipartObjectMD['x-amz-tagging'] = params.tagging;
         }
-        if (params.retentionInfo) {
-            multipartObjectMD.retentionInfo = params.retentionInfo;
+        if (params.retentionMode && params.retentionDate) {
+            multipartObjectMD.retentionMode = params.retentionMode;
+            multipartObjectMD.retentionDate = params.retentionDate;
```
Comment (ilkescality) [inline lib/services.js:396]: maybe setter could be used here as well; 

```suggestion
            multipartObjectMD.setRetentionMode(retentionMode) = params.retentionMode;
            multipartObjectMD.setRetentionDate(retentionDate) = params.retentionDate;
```

**Review by rahulreddy — APPROVED** [review]

File: yarn.lock [hunk 3]
```diff
@@ -264,14 +264,14 @@ arsenal@scality/Arsenal#9f2e74e:
...
   dependencies:
-    "@hapi/joi" "^15.1.0"
     JSONStream "^1.0.0"
     ajv "4.10.0"
     async "~2.1.5"
     debug "~2.3.3"
     diskusage "^1.1.1"
     ioredis "4.9.5"
     ipaddr.js "1.2.0"
+    joi "^10.6"
```
Comment (rahulreddy) [inline yarn.lock:274]: I wonder why this change is happening...`@hapi/joi` would be the right dependency

Comment (dora-korpar) [reply yarn.lock:274]: 🤔 not sure, that's just the auto-generated yarn.lock

File: lib/services.js (hunk 2 above)
Comment (dora-korpar) [reply lib/services.js:396]: Yeah, our use of objectMD as an instance vs as JSON is super inconsistent throughout Cloudserver, so I just followed what was done in the function around it. Definitely something to address though.

Comment (ilkescality) [reply lib/services.js:396]: Oh I see, sure, makes sense. :)

**Comment by rahulreddy** [comment]
I issued bypass as the test failure was related to Azure flakiness
